relayer/processor/message_processor.go

```diff
@@ -354,13 +371,27 @@ func (mp *messageProcessor) sendBatchMessages(
 	broadcastCtx, cancel := context.WithTimeout(ctx, messageSendTimeout)
 	defer cancel()
 
-	// messages are batch with appended MsgUpdateClient
-	msgs := make([]provider.RelayerMessage, 1+len(batch))
-	msgs[0] = mp.msgUpdateClient
-	fields := []zapcore.Field{}
-	for i, t := range batch {
-		msgs[i+1] = t.assembledMsg()
-		fields = append(fields, zap.Object(fmt.Sprintf("msg_%d", i), t))
+	var (
+		msgs   []provider.RelayerMessage
+		fields []zapcore.Field
+	)
+
+	if mp.localhost {
+		msgs = make([]provider.RelayerMessage, len(batch))
+		fields = []zapcore.Field{}
```

Comment:
Unnecessary assignment, since `append` will do the right thing with a nil slice. Same for the `else` block.